src/VideoPlayer/index.js

```diff
@@ -84,6 +95,14 @@ const fireOnConsumer = (event, args) => {
     consumer.fire('$videoPlayer' + event, args, videoEl.currentTime)
     consumer.fire('$videoPlayerEvent', event, args, videoEl.currentTime)
   }
+  if (event === events['timeupdate'] && subtitles.hasSubtitles) {
+    subtitles.currentSubtitle = SubtitlesParser.getSubtitleByTimeIndex(videoEl.currentTime)
+    if (subtitles.previousSubtitle !== subtitles.currentSubtitle) {
+      subtitles.previousSubtitle = subtitles.currentSubtitle
+      // firing SubtitleTextChanged event on consumer if text is changed
+      fireOnConsumer('SubtitleTextChanged', subtitles.currentSubtitle)
+    }
+  }
```

Comment:
I think this could be cleaner, since you're recursively calling itself now, but let's leave it like this for now